src/dfu.c

```diff
@@ -99,9 +108,10 @@ static int32_t thingset_dfu_boot()
         return err;
     }
 
-    LOG_INF("DFU finished, rebooting...");
+    LOG_INF("DFU finished, scheduling reboot...");
 
-    sys_reboot(SYS_REBOOT_COLD);
+    /* Schedule the reboot work to be executed after 1 second */
+    k_work_schedule(&reboot_work, K_SECONDS(1));
```

Comment:
We should add another check for `dfu_initialized` at the beginning of this function (same as for `thingset_dfu_write`). Otherwise we may call the function for an un-initialized work item.

Or alternatively, create and init the work item using `K_WORK_DELAYABLE_DEFINE` macro: `https://docs.zephyrproject.org/latest/kernel/services/threads/workqueue.html#c.K_WORK_DELAYABLE_DEFINE